core/src/test/java/org/infinispan/test/fwk/TestApplicationMetricsRegistry.java

```diff
@@ -0,0 +1,27 @@
+package org.infinispan.test.fwk;
+
+import org.eclipse.microprofile.metrics.MetricRegistry;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.metrics.impl.ApplicationMetricsRegistry;
+import org.infinispan.test.TestingUtil;
+
+import io.smallrye.metrics.MetricsRegistryImpl;
+
+/**
+ * Allows each cache manager to have its own separate MetricRegistry (the eclipse microprofile metrics standard uses a
+ * shared one per JVM).
+ *
+ * @author anistor@redhat.com
+ * @since 10.1
+ */
+final class TestApplicationMetricsRegistry extends ApplicationMetricsRegistry {
+
+   @Override
+   protected MetricRegistry makeRegistry() {
+      return new MetricsRegistryImpl();
+   }
+
+   static void replaceMetricsRegistry(DefaultCacheManager cacheManager) {
+      TestingUtil.replaceComponent(cacheManager, ApplicationMetricsRegistry.class, new TestApplicationMetricsRegistry(), true);
```

Comment:
Aren't some metrics already registered by the time you replace the component here?

I was really hoping you'd fix `GlobalConfigurationBuilder.addModule()` to not replace existing modules and to use `global.addModule(TestGlobalConfigurationBuilder.class).testGlobalComponent(ApplicationMetricsRegistry.class, new TestApplicationMetricsRegistry())` instead.

Reply:
done